soc/riscv/riscv-privilege/hpmicro/start.S

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/toolchain.h>
+#include <soc.h>
+#include "hpm_csr_regs.h"
+
+GTEXT(_start)
+
+SECTION_FUNC(init, _start)
+	/* Disable linker relaxation before GP register initialization. */
+	.option push
+	.option norelax
+	/* reset mstatus to 0*/
+	csrrw x0, mstatus, x0
+
+#ifdef __nds_execit
+	/* Initialize EXEC.IT table */
+	la t0, _ITB_BASE_
+	csrw uitb, t0
+#endif
+
+#ifdef __riscv_flen
+	/* Enable FPU */
+	li t0, CSR_MSTATUS_FS_MASK
+	csrrs t0, mstatus, t0
+
+	/* Initialize FCSR */
+	fscsr zero
+#endif
+	call _init_noncache
+
+	li t0, 0xf4002800
+	li t1, 0x3402
```

Comment:
Explain in comments what these values are.

Reply:
Is Wi-Fi enabled by default in this SoC, or is it enabled by something else? Also, would be nice if these values were defined somehow instead of being plain literals.

Reply:
BTW I meant that I want this explanation to be in this file as a comment, not in this thread :)

Reply:
Removed this code, called the `hpm_sdk` interface in the clock driver for configuration, and added comments.